lib/src/main/java/org/team100/lib/localization/NotePosition24ArrayListener.java

```diff
@@ -43,8 +48,22 @@ void consumeValues(NetworkTableEvent e) {
             } catch (RuntimeException ex) {
                 return;
             }
-
+            notes = new Optional[] { Optional.empty() };
+            int noteNum = 0;
+            Optional<Double> y = Optional.empty();
+            Optional<Double> x = Optional.empty();
+            Transform3d cameraInRobotCoordinates = Camera.get(fields[1]).getOffset();
+            CameraAngles ed = new CameraAngles(cameraInRobotCoordinates);
             for (NotePosition24 position : positions) {
+                if (position.getPitch() < Math.PI / 2 - cameraInRobotCoordinates.getRotation().getY()) {
```

Comment:
i'd suggest extracting this little bit of geometry to a utility class like PoseEstimatorHelper or GeometryUtil, so you can write some tests to verify it.